### ISSUE TITLE
Error 500 from records API when in violation of a not-null constraint

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#859]
## Reproduce

1. Use the front end
1. Open a table where at least one column has a NOT NULL constraint set
1. Click the `+` icon at the bottom left of the table to begin adding a new record.
1. Observe a request to `http://localhost:8000/api/v0/tables/24/records/` that 
1. Observe HTTP response code of 500. The response needs to conform to our error structure, to-be-decided in #560. In particular, for #775 I need a structured machine-readable way to know which column(s) were in violation of the constraint.

Related to #858 
Blocked by #760

**Comment by kgodey** [comment]
Moving this to the "Working with Tables" milestone since it blocks #775.

**Comment by seancolsen** [comment]
Marking this is unblocked since #560 is complete.

**Comment by dmos62** [comment]
Unless we split this into two issues, this is still blocked. The two issues would be:

1. handle the input, so that we would return an HTTP 4xx instead of 500;
2. adhere the API response to the new error spec.

Task 2 is still blocked pending error reporting implementation/concretization tracked in https://github.com/centerofci/mathesar/issues/883.

**Comment by seancolsen** [comment]
@dmos62 As I see it, #883 is a large issue that requires many small tasks, one of which would be closing this ticket. To put that another way, to close #883 we'll need to make sure all our endpoints, and all the HTTP methods, and all error scenarios adhere to to the new structure. That's potentially a lot of small independent tasks. This ticket is tracking one error scenario for one HTTP method for one endpoint. And unlike a lot of the other small tasks within #883, this one is blocking a front end ticket, so I think it's worth keeping this ticket separate from #883 and prioritizing it over all the other small tasks which #883 encompasses. No need to close #883 before starting work on this ticket.
